qa/L0_client_build_variants/test.sh

```diff
@@ -0,0 +1,245 @@
+    apt-add-repository 'deb https://apt.kitware.com/ubuntu/ bionic main' && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+    cmake-data=3.18.4-0kitware1 cmake=3.18.4-0kitware1; \
+else \
+    echo "ERROR: Only support CMAKE_UBUNTU_VERSION to be 18.04 or 20.04" && false; \
+fi && \
+cmake --version
+
+set +e
+
+mkdir -p /workspace/build
+
+#
+# Build without GPU support
+#
+(cd /workspace/build && \
+        rm -fr cc-clients java-clients python-clients && \
+        cmake -DCMAKE_INSTALL_PREFIX=/workspace/install \
+              -DTRITON_ENABLE_CC_HTTP=ON \
+              -DTRITON_ENABLE_CC_GRPC=ON \
+              -DTRITON_ENABLE_PYTHON_HTTP=ON \
+              -DTRITON_ENABLE_PYTHON_GRPC=ON \
+              -DTRITON_ENABLE_JAVA_HTTP=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_C_API=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_TFS=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_TS=ON \
+              -DTRITON_ENABLE_EXAMPLES=ON \
+              -DTRITON_ENABLE_TESTS=ON \
+              -DTRITON_ENABLE_GPU=OFF \
+              /workspace/client && \
+        make -j16 cc-clients java-clients python-clients)
+if [ $? -eq 0 ]; then
+    echo -e "\n***\n*** No-GPU Passed\n***"
+else
+    echo -e "\n***\n*** No-GPU FAILED\n***"
+    exit 1
+fi
+
+#
+# Build without HTTP
+# Skip this test for java-clients because we can only build
+# java-clients with http protocol
+#
+(cd /workspace/build && \
+        rm -fr cc-clients python-clients && \
+        cmake -DCMAKE_INSTALL_PREFIX=/workspace/install \
+              -DTRITON_ENABLE_CC_HTTP=OFF \
+              -DTRITON_ENABLE_CC_GRPC=ON \
+              -DTRITON_ENABLE_PYTHON_HTTP=OFF \
+              -DTRITON_ENABLE_PYTHON_GRPC=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_C_API=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_TFS=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_TS=ON \
+              -DTRITON_ENABLE_EXAMPLES=ON \
+              -DTRITON_ENABLE_TESTS=ON \
+              -DTRITON_ENABLE_GPU=ON \
+              /workspace/client && \
+        make -j16 cc-clients python-clients)
+if [ $? -eq 0 ]; then
+    echo -e "\n***\n*** No-HTTP Passed\n***"
+else
+    echo -e "\n***\n*** No-HTTP FAILED\n***"
+    exit 1
+fi
+
+#
+# Build without GRPC
+# Skip this test for java-clients because grpc protocol is not supported
+#
+(cd /workspace/build && \
+        rm -fr cc-clients python-clients && \
+        cmake -DCMAKE_INSTALL_PREFIX=/workspace/install \
+              -DTRITON_ENABLE_CC_HTTP=ON \
+              -DTRITON_ENABLE_CC_GRPC=OFF \
+              -DTRITON_ENABLE_PYTHON_HTTP=ON \
+              -DTRITON_ENABLE_PYTHON_GRPC=OFF \
+              -DTRITON_ENABLE_PERF_ANALYZER=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_C_API=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_TFS=ON \
+              -DTRITON_ENABLE_PERF_ANALYZER_TS=ON \
```

Comment:
How long does it takes to run the whole test? For each variant you set to build with TFS / TS which seems to be time consuming. It will probably be better to set them to OFF for most variants except for the ones that we actually wants to build with TFS / TS

Reply:
The whole test takes around 35 minutes on the CI. Thanks for the suggestion. I think it's better if we can reduce the time. Will make changes for that.

Reply:
Updated, please review. Thanks